extensions/ql-vscode/test/factories/db-item-factories.ts

```diff
@@ -1,34 +1,167 @@
 import {
   DbItemKind,
+  LocalDatabaseDbItem,
+  LocalListDbItem,
+  RemoteOwnerDbItem,
+  RemoteRepoDbItem,
+  RemoteSystemDefinedListDbItem,
   RemoteUserDefinedListDbItem,
+  RootLocalDbItem,
   RootRemoteDbItem,
 } from "../../src/databases/db-item";
 
-export function createRootRemoteDbItem(): RootRemoteDbItem {
+// Root Remote Db Items
+export function createRootRemoteDbItem({
+  children = [],
+  expanded = false,
+}: {
+  children?: Array<
+    | RemoteOwnerDbItem
+    | RemoteSystemDefinedListDbItem
+    | RemoteUserDefinedListDbItem
+  >;
+  expanded?: boolean;
+}): RootRemoteDbItem {
```

Comment:
You can also initialise the object we receive as a parameter to make that optional. E.g.

```ts
export function createRootRemoteDbItem({
  children = [],
  expanded = false,
}: {
  children?: RemoteDbItem[];
  expanded?: boolean;
} = {}): RootRemoteDbItem {
  return {
    kind: DbItemKind.RootRemote,
    children,
    expanded,
  };
}
```

That would mean you can do `createRootRemoteDbItem()` instead of `createRootRemoteDbItem({})`;

I think we should apply that to all factory functions. What do you think?

Reply:
Good point, didn't know that was possible!